bindings/rust/standard/bench/src/s2n_tls.rs

```diff
@@ -306,9 +282,46 @@ impl TlsConnection for S2NConnection {
         while read_offset < data_len {
             match self.connection.poll_recv(data) {
                 Poll::Ready(bytes_read) => read_offset += bytes_read?,
-                Poll::Pending => return Err("unexpected pending".into()),
+                Poll::Pending => return Err("blocking".into()),
             }
         }
         Ok(())
     }
+    
+    fn send_shutdown(&mut self) {
+        // This will return `Ready` if the peer already closed their connection.
+        // It will return `Pending` if the peer hasn't responded yet. 
+        let _ = self.connection.poll_shutdown();
```

Comment:
Also might not work for real networks. Could also double as "recv_shutdown" if the peer has already sent their close_notify, which would make the name misleading. "s2n_shutdown_send" might be more what you want here: https://github.com/aws/s2n-tls/blob/main/api/s2n.h#L2334-L2360